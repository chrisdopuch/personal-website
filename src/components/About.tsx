import Paper from '@material-ui/core/Paper';
import { Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component } from 'react';

const stylesDeclarations = (theme: Theme) => ({
  root: {
    paddingBottom: theme.spacing.unit * 2,
    paddingTop: theme.spacing.unit * 2,
    ...theme.mixins.gutters(),
  },
});

interface IAboutProps extends WithStyles<typeof stylesDeclarations> {}

class About extends Component<IAboutProps> {
  public render() {
    const { classes } = this.props;

    return (
      <div>
        <Paper className={classes.root} elevation={1}>
          <Typography variant="h5" component="h3">
            Who is Chris Dopuch?
          </Typography>
          <Typography component="p">
            Paper can be used to build surface or other elements for your application.
          </Typography>
        </Paper>
      </div>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(About);
