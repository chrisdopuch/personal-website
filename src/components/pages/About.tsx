import Avatar from '@material-ui/core/Avatar';
import Divider from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import Paper from '@material-ui/core/Paper';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import Typography from '@material-ui/core/Typography';
import BuildIcon from '@material-ui/icons/Build';
import CodeIcon from '@material-ui/icons/Code';
import HomeIcon from '@material-ui/icons/Home';
import SchoolIcon from '@material-ui/icons/School';
import WorkIcon from '@material-ui/icons/Work';
import React, { Component, ComponentClass, ComponentType, Fragment } from 'react';
import meSinging from '../../images/me-singing.jpg';

const stylesDeclarations = (theme: Theme) =>
  createStyles({
    avatar: {
      height: '27vw',
      margin: 20,
      width: '27vw',
    },
    centered: {
      display: 'flex',
      justifyContent: 'center',
    },
    divider: {
      marginBottom: '10px',
      marginTop: '10px',
    },
    gridContainer: {
      marginTop: '10px',
    },
    list: {
      backgroundColor: theme.palette.background.paper,
      marginTop: '10px',
    },
    root: {
      paddingBottom: theme.spacing.unit * 2,
      paddingLeft: theme.spacing.unit * 2,
      paddingRight: theme.spacing.unit * 3,
      paddingTop: theme.spacing.unit * 3,
      ...theme.mixins.gutters(),
    },
  });

interface IQuickFact {
  Icon: ComponentType<SvgIconProps>;
  primary: string;
  secondary: string;
  tertiary?: string;
  component?: ComponentClass<any> | string;
  componentProps?: object;
}

interface IAboutProps extends WithStyles<typeof stylesDeclarations> {}

export class About extends Component<IAboutProps> {
  public constructor(props: IAboutProps) {
    super(props);
    this.renderHeader = this.renderHeader.bind(this);
    this.renderAvatar = this.renderAvatar.bind(this);
    this.renderAvatar = this.renderAvatar.bind(this);
    this.renderQuickFacts = this.renderQuickFacts.bind(this);
  }

  public renderHeader() {
    const { classes } = this.props;
    return (
      <header>
        <Typography variant="h2" component="h3">
          About me
        </Typography>
        <Divider className={classes.divider} />
        <Typography component="p">
          Hi, my name is Chris Dopuch and I'm a Software Engineer working in Portland, Oregon. I moved to Portland from
          Columbia, Missouri after graduating from University of Missouri with both a CS and an IT degree. Currently I
          am employed by Moovel North America where I work on web applications and services using technologies like
          NodeJS, Typescript, React, and Redux. Outside of work I enjoy tabletop and roleplay gaming, watching cult
          movies, going to heavy metal concerts, drawing, and painting. On this site you can find many examples of my
          work including my art gallery as well as many of my programming projects.
        </Typography>
      </header>
    );
  }

  public renderAvatar() {
    const { classes } = this.props;
    return (
      <div className={classes.centered}>
        <Avatar alt="Chris Dopuch" src={meSinging} className={classes.avatar} />
      </div>
    );
  }

  public renderQuickFacts() {
    const { classes } = this.props;
    const quickFacts: IQuickFact[] = [
      {
        Icon: HomeIcon,
        primary: 'Location',
        secondary: 'Portland, Oregon',
      },
      {
        Icon: WorkIcon,
        primary: 'Job Title',
        secondary: 'Software Engineer at Moovel North America',
      },
      {
        Icon: SchoolIcon,
        primary: 'Education',
        secondary: 'BA in Computer Science, Univeristy of Missouri',
        tertiary: 'BS in Information Technology, Univeristy of Missouri',
      },
      {
        Icon: BuildIcon,
        primary: 'Technologies',
        secondary: 'NodeJS, Typescript, React, Redux, Express, HapiJS',
        tertiary: 'Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP',
      },
      {
        Icon: CodeIcon,
        component: 'a',
        componentProps: { href: 'http://github.com/chrisdopuch' },
        primary: 'Github Profile',
        secondary: 'http://github.com/chrisdopuch',
      },
    ];
    return (
      <Fragment>
        <Typography variant="h5" component="h5">
          Quick Facts
        </Typography>
        <Divider className={classes.divider} />
        <div className={classes.list}>
          <List>
            {quickFacts.map(({ primary, secondary, tertiary, Icon, component, componentProps }) => {
              return (
                <Fragment key={primary}>
                  <ListItem key={primary} component={component} {...componentProps} divider={!tertiary}>
                    <ListItemAvatar>
                      <Avatar>
                        <Icon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary={primary} secondary={secondary} />
                  </ListItem>
                  {tertiary ? (
                    <ListItem key={tertiary} divider={true}>
                      <ListItemText inset={true} secondary={tertiary} />
                    </ListItem>
                  ) : null}
                </Fragment>
              );
            })}
          </List>
        </div>
      </Fragment>
    );
  }

  public render() {
    const { classes } = this.props;

    return (
      <article>
        <Paper className={classes.root} elevation={1}>
          {this.renderHeader()}
          <Hidden smDown={true} implementation="css">
            <Grid container={true} spacing={24} className={classes.gridContainer}>
              <Grid item={true} xs={6}>
                {this.renderAvatar()}
              </Grid>
              <Grid item={true} xs={6}>
                {this.renderQuickFacts()}
              </Grid>
            </Grid>
          </Hidden>
          <Hidden mdUp={true} implementation="css">
            {this.renderAvatar()}
            {this.renderQuickFacts()}
          </Hidden>
        </Paper>
      </article>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(About);
