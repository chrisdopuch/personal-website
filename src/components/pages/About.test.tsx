import { shallow } from 'enzyme';
import { History, Location } from 'history';
import React from 'react';
import { match } from 'react-router';
import withPageView from '../../hooks/usePageView';
import { About, Header, ProfilePicture, QuickFacts } from './About';

jest.mock('../../hooks/usePageView', () => {
  return jest.fn();
});

const classes = {
  avatar: 'avatar',
  centered: 'centered',
  divider: 'divider',
  gridContainer: 'gridContainer',
  list: 'list',
  root: 'root',
};

const getDefaultProps = () => {
  return {
    classes,
    history: {} as History,
    location: {} as Location,
    match: {} as match,
  };
};

describe('About', () => {
  describe('Main Component', () => {
    it('should render with default props', () => {
      const props = getDefaultProps();
      const wrapper = shallow(<About {...props} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should call the usePageView hook', () => {
      const props = {
        ...getDefaultProps(),
        location: {
          pathname: '/foo',
          search: '?bar=baz',
        } as Location,
      };
      shallow(<About {...props} />);

      expect(withPageView).toBeCalledWith(props.location);
    });
  });

  describe('ProfilePicture', () => {
    it('should render with default props', () => {
      const props = getDefaultProps();
      const wrapper = shallow(<ProfilePicture {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Header', () => {
    it('should render with default props', () => {
      const props = getDefaultProps();
      const wrapper = shallow(<Header {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('QuickFacts', () => {
    it('should render with default props', () => {
      const props = getDefaultProps();
      const wrapper = shallow(<QuickFacts {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
