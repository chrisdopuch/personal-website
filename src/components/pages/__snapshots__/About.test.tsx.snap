// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About #renderAvatar should render the avatar 1`] = `
<div
  className="centered"
>
  <WithStyles(Avatar)
    alt="Chris Dopuch"
    className="avatar"
    src="test-file-stub"
  />
</div>
`;

exports[`About #renderHeader should render the header 1`] = `
<header>
  <WithStyles(Typography)
    component="h3"
    variant="h2"
  >
    About me
  </WithStyles(Typography)>
  <WithStyles(Divider)
    className="divider"
  />
  <WithStyles(Typography)
    component="p"
  >
    Hi, my name is Chris Dopuch and I'm a Software Engineer working in Portland, Oregon. I moved to Portland from Columbia, Missouri after graduating from University of Missouri with both a CS and an IT degree. Currently I am employed by Moovel North America where I work on web applications and services using technologies like NodeJS, Typescript, React, and Redux. Outside of work I enjoy tabletop and roleplay gaming, watching cult movies, going to heavy metal concerts, drawing, and painting. On this site you can find many examples of my work including my art gallery as well as many of my programming projects.
  </WithStyles(Typography)>
</header>
`;

exports[`About #renderQuickFacts should render the quick facts list 1`] = `
<React.Fragment>
  <WithStyles(Typography)
    component="h5"
    variant="h5"
  >
    Quick Facts
  </WithStyles(Typography)>
  <WithStyles(Divider)
    className="divider"
  />
  <div
    className="list"
  >
    <WithStyles(List)>
      <React.Fragment>
        <WithStyles(ListItem)
          divider={true}
        >
          <WithStyles(ListItemAvatar)>
            <WithStyles(Avatar)>
              <pure(HomeIcon) />
            </WithStyles(Avatar)>
          </WithStyles(ListItemAvatar)>
          <WithStyles(ListItemText)
            primary="Location"
            secondary="Portland, Oregon"
          />
        </WithStyles(ListItem)>
      </React.Fragment>
      <React.Fragment>
        <WithStyles(ListItem)
          divider={true}
        >
          <WithStyles(ListItemAvatar)>
            <WithStyles(Avatar)>
              <pure(WorkIcon) />
            </WithStyles(Avatar)>
          </WithStyles(ListItemAvatar)>
          <WithStyles(ListItemText)
            primary="Job Title"
            secondary="Software Engineer at Moovel North America"
          />
        </WithStyles(ListItem)>
      </React.Fragment>
      <React.Fragment>
        <WithStyles(ListItem)
          divider={false}
        >
          <WithStyles(ListItemAvatar)>
            <WithStyles(Avatar)>
              <pure(SchoolIcon) />
            </WithStyles(Avatar)>
          </WithStyles(ListItemAvatar)>
          <WithStyles(ListItemText)
            primary="Education"
            secondary="BA in Computer Science, Univeristy of Missouri"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          divider={true}
        >
          <WithStyles(ListItemText)
            inset={true}
            secondary="BS in Information Technology, Univeristy of Missouri"
          />
        </WithStyles(ListItem)>
      </React.Fragment>
      <React.Fragment>
        <WithStyles(ListItem)
          divider={false}
        >
          <WithStyles(ListItemAvatar)>
            <WithStyles(Avatar)>
              <pure(BuildIcon) />
            </WithStyles(Avatar)>
          </WithStyles(ListItemAvatar)>
          <WithStyles(ListItemText)
            primary="Technologies"
            secondary="NodeJS, Typescript, React, Redux, Express, HapiJS"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          divider={true}
        >
          <WithStyles(ListItemText)
            inset={true}
            secondary="Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP"
          />
        </WithStyles(ListItem)>
      </React.Fragment>
      <React.Fragment>
        <WithStyles(ListItem)
          component="a"
          divider={true}
          href="http://github.com/chrisdopuch"
        >
          <WithStyles(ListItemAvatar)>
            <WithStyles(Avatar)>
              <pure(CodeIcon) />
            </WithStyles(Avatar)>
          </WithStyles(ListItemAvatar)>
          <WithStyles(ListItemText)
            primary="Github Profile"
            secondary="http://github.com/chrisdopuch"
          />
        </WithStyles(ListItem)>
      </React.Fragment>
    </WithStyles(List)>
  </div>
</React.Fragment>
`;

exports[`About Component should render with default props 1`] = `
<article>
  <WithStyles(Paper)
    className="root"
    elevation={1}
  >
    <header>
      <WithStyles(Typography)
        component="h3"
        variant="h2"
      >
        About me
      </WithStyles(Typography)>
      <WithStyles(Divider)
        className="divider"
      />
      <WithStyles(Typography)
        component="p"
      >
        Hi, my name is Chris Dopuch and I'm a Software Engineer working in Portland, Oregon. I moved to Portland from Columbia, Missouri after graduating from University of Missouri with both a CS and an IT degree. Currently I am employed by Moovel North America where I work on web applications and services using technologies like NodeJS, Typescript, React, and Redux. Outside of work I enjoy tabletop and roleplay gaming, watching cult movies, going to heavy metal concerts, drawing, and painting. On this site you can find many examples of my work including my art gallery as well as many of my programming projects.
      </WithStyles(Typography)>
    </header>
    <Hidden
      implementation="css"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={false}
      smDown={true}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Grid)
        className="gridContainer"
        container={true}
        spacing={24}
      >
        <WithStyles(Grid)
          item={true}
          xs={6}
        >
          <div
            className="centered"
          >
            <WithStyles(Avatar)
              alt="Chris Dopuch"
              className="avatar"
              src="test-file-stub"
            />
          </div>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={6}
        >
          <WithStyles(Typography)
            component="h5"
            variant="h5"
          >
            Quick Facts
          </WithStyles(Typography)>
          <WithStyles(Divider)
            className="divider"
          />
          <div
            className="list"
          >
            <WithStyles(List)>
              <WithStyles(ListItem)
                divider={true}
                key="Location"
              >
                <WithStyles(ListItemAvatar)>
                  <WithStyles(Avatar)>
                    <pure(HomeIcon) />
                  </WithStyles(Avatar)>
                </WithStyles(ListItemAvatar)>
                <WithStyles(ListItemText)
                  primary="Location"
                  secondary="Portland, Oregon"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                divider={true}
                key="Job Title"
              >
                <WithStyles(ListItemAvatar)>
                  <WithStyles(Avatar)>
                    <pure(WorkIcon) />
                  </WithStyles(Avatar)>
                </WithStyles(ListItemAvatar)>
                <WithStyles(ListItemText)
                  primary="Job Title"
                  secondary="Software Engineer at Moovel North America"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                divider={false}
                key="Education"
              >
                <WithStyles(ListItemAvatar)>
                  <WithStyles(Avatar)>
                    <pure(SchoolIcon) />
                  </WithStyles(Avatar)>
                </WithStyles(ListItemAvatar)>
                <WithStyles(ListItemText)
                  primary="Education"
                  secondary="BA in Computer Science, Univeristy of Missouri"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                divider={true}
                key="BS in Information Technology, Univeristy of Missouri"
              >
                <WithStyles(ListItemText)
                  inset={true}
                  secondary="BS in Information Technology, Univeristy of Missouri"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                divider={false}
                key="Technologies"
              >
                <WithStyles(ListItemAvatar)>
                  <WithStyles(Avatar)>
                    <pure(BuildIcon) />
                  </WithStyles(Avatar)>
                </WithStyles(ListItemAvatar)>
                <WithStyles(ListItemText)
                  primary="Technologies"
                  secondary="NodeJS, Typescript, React, Redux, Express, HapiJS"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                divider={true}
                key="Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP"
              >
                <WithStyles(ListItemText)
                  inset={true}
                  secondary="Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP"
                />
              </WithStyles(ListItem)>
              <WithStyles(ListItem)
                component="a"
                divider={true}
                href="http://github.com/chrisdopuch"
                key="Github Profile"
              >
                <WithStyles(ListItemAvatar)>
                  <WithStyles(Avatar)>
                    <pure(CodeIcon) />
                  </WithStyles(Avatar)>
                </WithStyles(ListItemAvatar)>
                <WithStyles(ListItemText)
                  primary="Github Profile"
                  secondary="http://github.com/chrisdopuch"
                />
              </WithStyles(ListItem)>
            </WithStyles(List)>
          </div>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </Hidden>
    <Hidden
      implementation="css"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={true}
      smDown={false}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <div
        className="centered"
      >
        <WithStyles(Avatar)
          alt="Chris Dopuch"
          className="avatar"
          src="test-file-stub"
        />
      </div>
      <WithStyles(Typography)
        component="h5"
        variant="h5"
      >
        Quick Facts
      </WithStyles(Typography)>
      <WithStyles(Divider)
        className="divider"
      />
      <div
        className="list"
      >
        <WithStyles(List)>
          <WithStyles(ListItem)
            divider={true}
            key="Location"
          >
            <WithStyles(ListItemAvatar)>
              <WithStyles(Avatar)>
                <pure(HomeIcon) />
              </WithStyles(Avatar)>
            </WithStyles(ListItemAvatar)>
            <WithStyles(ListItemText)
              primary="Location"
              secondary="Portland, Oregon"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            divider={true}
            key="Job Title"
          >
            <WithStyles(ListItemAvatar)>
              <WithStyles(Avatar)>
                <pure(WorkIcon) />
              </WithStyles(Avatar)>
            </WithStyles(ListItemAvatar)>
            <WithStyles(ListItemText)
              primary="Job Title"
              secondary="Software Engineer at Moovel North America"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            divider={false}
            key="Education"
          >
            <WithStyles(ListItemAvatar)>
              <WithStyles(Avatar)>
                <pure(SchoolIcon) />
              </WithStyles(Avatar)>
            </WithStyles(ListItemAvatar)>
            <WithStyles(ListItemText)
              primary="Education"
              secondary="BA in Computer Science, Univeristy of Missouri"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            divider={true}
            key="BS in Information Technology, Univeristy of Missouri"
          >
            <WithStyles(ListItemText)
              inset={true}
              secondary="BS in Information Technology, Univeristy of Missouri"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            divider={false}
            key="Technologies"
          >
            <WithStyles(ListItemAvatar)>
              <WithStyles(Avatar)>
                <pure(BuildIcon) />
              </WithStyles(Avatar)>
            </WithStyles(ListItemAvatar)>
            <WithStyles(ListItemText)
              primary="Technologies"
              secondary="NodeJS, Typescript, React, Redux, Express, HapiJS"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            divider={true}
            key="Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP"
          >
            <WithStyles(ListItemText)
              inset={true}
              secondary="Go, Ruby on Rails, Swagger, LESS, SASS, Webpack, PHP"
            />
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
            component="a"
            divider={true}
            href="http://github.com/chrisdopuch"
            key="Github Profile"
          >
            <WithStyles(ListItemAvatar)>
              <WithStyles(Avatar)>
                <pure(CodeIcon) />
              </WithStyles(Avatar)>
            </WithStyles(ListItemAvatar)>
            <WithStyles(ListItemText)
              primary="Github Profile"
              secondary="http://github.com/chrisdopuch"
            />
          </WithStyles(ListItem)>
        </WithStyles(List)>
      </div>
    </Hidden>
  </WithStyles(Paper)>
</article>
`;
